docs/proposals/04-etcd-member-custom-resource.md

```diff
@@ -0,0 +1,561 @@
+      endRevision: <end revision of etcd db captured in the snapshot>
+    accumulatedDeltaSize: <total size of delta snapshots since last full snapshot>
+  restorations:
+  - type: <FromSnapshot | FromLeader>
+    status: <Failed | Success | In-Progress>
+    startTime: <start time of restoration>
+    endTime: <end time of restoration>
+  defragmentations:
+  - startTime: <start time of defragmentation>
+    endTime: <end time of defragmentation>
+    reason: 
+    message:
+    initialDBSize: <size of etcd DB prior to defragmentation>
+    finalDBSize: <size of etcd DB post degragmentation>
+  volumeMismatches:
+  - identifiedAt: <time at which wrong volume mount was identified>
+    fixedAt: <time at which correct volume was mounted>
+    volumeID: <volume ID of wrong volume that got mounted>
+    numRestarts: <num of pod restarts that were attempted>
+  transitions:
+  - state: <name of the state that the etcd-member has transitioned to>
+    subState: <name of the sub-state if any>
+    reason: <reason code for the transition>
+    transitionTime: <time of transition to this state>
+    message: <detailed message if any>
+```
+
+### Lifecycle of an EtcdMember
+
+#### Creation
+
+Druid creates an `EtcdMember` resource for every replica in `etcd.Spec.Replicas` during reconciliation of an etcd resource. For a fresh etcd cluster this is done prior to creation of the StatefulSet resource and for an existing cluster which has now been scaled-up, it is done prior to updating the StatefulSet resource.
+
+#### Updation
+
+All fields in `EtcdMember.Status` are *only* updated by the corresponding etcd-member. Druid only consumes the information published via `EtcdMember` resources.
+
+#### Deletion
+
+Druid is responsible for deletion of all existing `EtcdMember` resources for an etcd cluster. There are three scenarios where an `EtcdMember` resource will be deleted:
+
+1. Deletion of etcd resource.
+
+2. Scale down of an etcd cluster to 0 replicas due to hibernation of the k8s control plane.
+
+3. Transient scale down of an etcd cluster to 0 replicas to recover from a quorum loss.
+
+Authors found no reason to retain EtcdMember resources when the etcd cluster is scale down to 0 replicas since the information contained in each EtcdMember resource would no longer represent the current state of each member and would thus be stale. Any controller in druid which acts upon the `EtcdMember.Status` could potentially take incorrect actions.
+
+#### Reconciliation
+
+Authors propose to introduce a new controller (lets call it `etcd-member-controller`) which watches for changes to the `EtcdMember` resource(s). If a reconciliation of an `Etcd` resource is required as a result of change in `EtcdMember` status then this controller should enqueue an event and force a reconciliation via existing `etcd-controller`, thus preserving the single-actor-principal constraint which ensures deterministic changes to etcd cluster resources.
```

Comment:
Just curious what happens to the[ EtcdMemberStatus](https://github.com/gardener/etcd-druid/blob/9a598d05e639099ddf404803f87376852261a052/api/v1alpha1/types_etcd.go#L358) which is currently managed by druid in the `etcd` resource. 
Do we continue to maintain it there as well, or will  the new `EtcdMember` be the only place to check this. 
May be a few sentences here might help to gain some context and if there is an impact of this proposal on the existing functionality as currently it focusses only on `EtcdMember`

Reply:
No, it will only have references to `EtcdMember` resources that are created for an `Etcd` resource as part of its status. I have now added proposed changes to `Etcd` resource in a new section under `API`.

Reply:
Thanks looks complete now with specifics of etcd api changes.